From: George Moussalem <george.moussalem@outlook.com>
Date: Wed, 28 May 2025 08:37:28 +0400
Subject: [PATCH] arm64: dts: qcom: ipq5018: Add vendor compatible to mdio bus

QCA SSDK needs to register the MDIO bus in its driver. Without a proper reference,
it will crash. Since it tries to look up the MDIO bus using compatible strings that
aren't upstreamed, add the vendor compatible.

Signed-off-by: George Moussalem <george.moussalem@outlook.com>
--- a/arch/arm64/boot/dts/qcom/ipq5018.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq5018.dtsi
@@ -201,7 +201,7 @@
 		};
 
 		mdio0: mdio@88000 {
-			compatible = "qcom,ipq5018-mdio";
+			compatible = "qcom,ipq5018-mdio", "qcom,ipq40xx-mdio";
 			reg = <0x00088000 0x64>,
 			      <0x019475c4 0x4>;
 			#address-cells = <1>;
